Refactor all post 1.5.7 funcs to use the new parser

Pirulax opened the pull request:
I've talked with Qais, and he said it would be a big help to refactor all these new functions.

Also, the *not done list*:

- dxDrawPrimitive3D  / dxDrawMaterialPrimitive3D - hard to read table structure. Its pretty specific, so i wont bother trying it with the new parser, will just add a big overhead.
- engineGetModelTextures  - performance issue(returning std::unordred_map, with possibly a lot of strings: a lot of HEAP allocs, which would kill the performance. I suggest to implement `std::vector<std::pair<const char*, Type*>>, or with string_view.`
- SetColShapeSize - argument types depends on colshape type - impossible to do with the new parser(well, not impossible, but im not sure what would be the best way, ill leave it up to you)
- GetColPolygonPoints - modifying it would kill the performance because of std::vector heap allocation on return
- [Add possibility of modifying dynamic objects behavior  (#784)](https://github.com/multitheftauto/mtasa-blue/commit/)

Also, maybe its not the best place to discuss: I think adding the ability to return `std::optional` to the new parser would be dope. Theres a function in the col shape file, where I've left a `TODO` comment, because its ugly and we need a refactor.

Pirulax committed:
refactor DxDrawWiredSphere to use the new parser

Pirulax committed:
Refactor EngineSetModelVisibleTime to use the new parser

Pirulax committed:
Refactor EngineGetModelVisibleTime, also add <optinal> and <variant> includes to cluaenginedefs.h

Pirulax committed:
Refactor DxGetTextSize to use the new parser. Add an overload to ResolveD3DXFont to accept std::variant, also move the old definition of it into CStaticFDs.h

Pirulax committed:
Refactor EngineResetModelLODDistance fucked commit

Pirulax committed:
Refactor GetPlayerScriptDebugLevel to use the new parser. Remove GetPlayerScriptDebugLevel from CStaticFunc

Pirulax committed:
Refactor SetPlayerScriptDebugLevel to use the new parser

Pirulax committed:
Refactor SetPedArmor to use the new parser. Remove SetPedArmor from CStaticFunc.

Pirulax committed:
Refactor SetPedFightingStyle to use the new parser.

Pirulax committed:
Refactor AreVehicleLightsOn to use the new parser

Pirulax committed:
Refactor all CLuaColShapeDefs functions in #19eb2e670 to use the new parser

Pirulax committed:
refactor isResourceProtected to use the new parser

Pirulax commented:
I know there are a few messed up commits. Im sorry.

Pirulax commented:
Seems like we either move 
`{"resetBlurLevel", CLuaDefs::ArgumentParser<CLuaFunctionDefs::ResetBlurLevel>},`  to `LuaDefs.World`(Somehow, because `CLuaManager` loads all the defs from where) or we make `CLuaDefs::ArgumentParser` functions public or we do a `private CLuaDefs` to `CLuaManager` which would be dumb.

Pirulax commented:
I also refactored the non OOP functions in ColShapeDefs to return numbers instead of vector(s)

StrixG commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -564,140 +539,62 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
-    uint             uiPointIndex;
-    CVector2D        vecPoint;
+    if (colShape->GetShapeType() != COLSHAPE_POLYGON)
+        throw std::invalid_argument("ColShape must be Polygon");
```
```suggestion
        throw std::invalid_argument("ColShape must be a polygon");
```

StrixG commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -564,140 +539,62 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
-    if (argStream.HasErrors())
-        return luaL_error(luaVM, argStream.GetFullErrorMessage());
+    if (!CStaticFunctionDefinitions::SetColPolygonPointPosition(static_cast<CClientColPolygon*>(colShape), pointIndex, point))
+        std::invalid_argument("Invalid points index");
```
```suggestion
        std::invalid_argument("Invalid point index");
```

StrixG commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -564,140 +539,62 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
-    uint             uiPointIndex;
-    CVector2D        vecPoint;
+    if (colShape->GetShapeType() != COLSHAPE_POLYGON)
+        throw std::invalid_argument("ColShape must be Polygon");
```
```suggestion
        throw std::invalid_argument("ColShape must be a polygon");
```

StrixG commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -564,140 +539,62 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
-    CClientColShape* pColShape;
-    uint             uiPointIndex;
+    if (colShape->GetShapeType() != COLSHAPE_POLYGON)
+        throw std::invalid_argument("ColShape must be Polygon");
```
```suggestion
        throw std::invalid_argument("ColShape must be a polygon");
```

StrixG commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -564,140 +539,62 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
 
-    argStream.SetCustomError("ColShape must be Polygon");
-    return luaL_error(luaVM, argStream.GetFullErrorMessage());
+    throw std::invalid_argument("Invalid points index");
```
```suggestion
    throw std::invalid_argument("Invalid point index");
```

Pirulax commented:
There you go.
The build doesnt compile. 
I need someone who could answer my previous question,

> Seems like we either move
> {"resetBlurLevel", CLuaDefs::ArgumentParser<CLuaFunctionDefs::ResetBlurLevel>}, to LuaDefs.World(Somehow, because CLuaManager loads all the defs from where) or we make CLuaDefs::ArgumentParser functions public or we do a private CLuaDefs to CLuaManager which would be dumb.

also, `DxDrawWiredSphere` has `const SColorARGB color`, how should we deal with that?

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -558,144 +533,66 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
-    argStream.ReadUserData(pColShape);
-    argStream.ReadNumber(uiPointIndex);
-    argStream.ReadVector2D(vecPoint);
+        pointIndex--; // lua starts at 1, we at 0
```
```suggestion
    pointIndex--; // lua starts at 1, we at 0
```

Should there be an extra indent here?

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.cpp:
```diff
@@ -2071,41 +2071,12 @@ int CLuaDrawingDefs::DxSetTextureEdge(lua_State* luaVM)
-    //  bool dxDrawWiredSphere( float x, float y, float z, float radius, color theColor, float fLineWidth, uint iterations )
-    CVector          vecPosition;
-    float            fRadius;
-    SColorARGB       color(64, 255, 0, 0);
-    float            fLineWidth = 1;
-    uint             uiIterations = 1;
```
Defaults are lost.

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.cpp:
```diff
@@ -2071,41 +2071,12 @@ int CLuaDrawingDefs::DxSetTextureEdge(lua_State* luaVM)
-    CGraphicsInterface* pGraphics = g_pCore->GetGraphics();
+    // Greater than 4, crash the game
+    if (iterations == 0 || iterations > 4)
+        throw std::invalid_argument("argument 'iterations' must be between 1 and 4");
```
```suggestion
        throw std::invalid_argument("Iterations must be between 1 and 4");
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -1104,38 +1104,26 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
-    argStream.ReadNumber(cHourOn);
-    argStream.ReadNumber(cHourOff);
+    if (hourOn > 24 || hourOn < 0)
+        throw std::invalid_argument("hour on must be between 0 and 24");
```
EngineSetModelVisibleTime

```suggestion
        throw std::invalid_argument("hourOn must be between 0 and 24");
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -1104,38 +1104,26 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
-            if (cHourOn > cHourOff)
-                std::swap(cHourOn, cHourOff);
+    if (hourOff > 24 || hourOff < 0)
+        throw std::invalid_argument("hour on must be between 0 and 24");
```
EngineSetModelVisibleTime

```suggestion
        throw std::invalid_argument("hourOff must be between 0 and 24");
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -1104,38 +1104,26 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
-    return 1;
+
+    // either convert string to modelid using CModelNames::ResolveModelID, or assign the id from the variant
+    const unsigned short modelID = std::holds_alternative<std::string>(variantModelID) ? CModelNames::ResolveModelID(std::get<std::string>(variantModelID)) : std::get<unsigned int>(variantModelID);
```
EngineSetModelVisibleTime

```suggestion
    unsigned short modelID;
    if (std::holds_alternative<std::string>(variantModelID))
        modelID = CModelNames::ResolveModelID(std::get<std::string>(variantModelID));
    else
        modelID = std::get<unsigned int>(variantModelID);
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -1104,38 +1104,26 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
+
+    CModelInfo* const modelInfo = g_pGame->GetModelInfo(modelID);
+    if (!modelInfo)
+        throw std::invalid_argument(SString("Can't find model id %u", modelID));
```
Do we want to use `std::format` instead? Needs second opinion.

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -1126,41 +1126,17 @@ bool CLuaEngineDefs::EngineSetModelVisibleTime(const std::variant<std::string, u
-    else
-        luaL_error(luaVM, argStream.GetFullErrorMessage());
+    // either convert string to modelid using CModelNames::ResolveModelID, or assign the id from the variant
+    const unsigned short modelID = std::holds_alternative<std::string>(variantModelID) ? CModelNames::ResolveModelID(std::get<std::string>(variantModelID)) : std::get<unsigned int>(variantModelID);
```
EngineGetModelVisibleTime

```suggestion
    unsigned short modelID;
    if (std::holds_alternative<std::string>(variantModelID))
        modelID = CModelNames::ResolveModelID(std::get<std::string>(variantModelID));
    else
        modelID = std::get<unsigned int>(variantModelID);

```

Pirulax replied:
I'll just overload the `CModelNames::ResolveModelID` function to take an `variant` as an argument.

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.h:
```diff
@@ -48,7 +50,7 @@ class CLuaEngineDefs : public CLuaDefs
     LUA_DECLARE(EngineGetVisibleTextureNames);
 
-    static bool EngineSetModelVisibleTime(const std::variant<std::string, unsigned int> variantModelID, char hourOn, char hourOff);
+    static bool EngineSetModelVisibleTime(lua_State* const luaVM, const std::variant<std::string, unsigned int> variantModelID, char hourOn, char hourOff);
```
EngineGetModelVisibleTime. This shouldn't be here

```suggestion
    static bool EngineSetModelVisibleTime(const std::variant<std::string, unsigned int> variantModelID, char hourOn, char hourOff);
```

Pirulax replied:
I think ive squashed them when rearranging the commits yesterday.

qaisjp commented on Client/mods/deathmatch/logic/CStaticFunctionDefinitions.h:
```diff
@@ -402,7 +402,21 @@ class CStaticFunctionDefinitions
+        return std::get<CClientDxFont*>(variant)->GetD3DXFont();
+    }
+
+    static inline  ID3DXFont* CStaticFunctionDefinitions::ResolveD3DXFont(eFontType fontType, CClientDxFont* pDxFontElement)
```
DxGetTextSize

```suggestion
    static inline ID3DXFont* CStaticFunctionDefinitions::ResolveD3DXFont(eFontType fontType, CClientDxFont* pDxFontElement)
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.h:
```diff
@@ -31,7 +33,39 @@ class CLuaDrawingDefs : public CLuaDefs
+    )
+    {
+        const auto size = OOP_DxGetTextSize(
+            optFont.has_value() ? std::move(optFont.value()) : FONT_DEFAULT,
```
DxGetTextSize

Not sure if this will work, but might be preferable

```suggestion
            std::move(optFont.value_or(FONT_DEFAULT)),
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.cpp:
```diff
@@ -854,98 +857,39 @@ int CLuaDrawingDefs::OOP_DxGetTextWidth(lua_State* luaVM)
-        fScaleX = vecScale.fX;
-        fScaleY = vecScale.fY;
+        scale.fX = optScaleXY.value();
+        scale.fY = optScaleY.has_value() ? optScaleY.value() : scale.fX;
```
DxGetTextSize
```suggestion
        scale.fY = optScaleY.value_or(scale.fX);
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.cpp:
```diff
@@ -854,98 +857,39 @@ int CLuaDrawingDefs::OOP_DxGetTextWidth(lua_State* luaVM)
+        scale.fY,
+        CStaticFunctionDefinitions::ResolveD3DXFont(variantFont),
+        optWordBreak.value_or(false),
+        optColorCoded.value_or(false)
```
DxGetTextSize

```suggestion
        optColorCoded.value_or(false),
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.h:
```diff
@@ -31,7 +33,39 @@ class CLuaDrawingDefs : public CLuaDefs
-    LUA_DECLARE_OOP(DxGetTextSize);
+
+    static CVector2D OOP_DxGetTextSize(
+        std::variant<CClientDxFont*, eFontType> font, // is optional, so we can call it from DxGetTextSize with a std::nullopt, and we just grab the FONT_DEFAULT.
```
DxGetTextSize
```suggestion
        // This is a variant for use by DxGetTextSize
        std::variant<CClientDxFont*, eFontType> font,
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -772,31 +772,21 @@ int CLuaEngineDefs::EngineSetModelLODDistance(lua_State* luaVM)
+    const unsigned short modelID = std::holds_alternative<std::string>(variantModelID) ? CModelNames::ResolveModelID(std::get<std::string>(variantModelID)) : std::get<unsigned int>(variantModelID);
+    CModelInfo* const modelInfo = g_pGame->GetModelInfo(modelID);
+    if (!modelInfo)
+        throw std::invalid_argument(SString("Invalid model id %u", modelID));
```
- Old code: just returns false
- New code: error

I think this should be a warning. Needs second opinion.

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -772,31 +772,21 @@ int CLuaEngineDefs::EngineSetModelLODDistance(lua_State* luaVM)
-    SString          strModel = "";
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadString(strModel);
+    const unsigned short modelID = std::holds_alternative<std::string>(variantModelID) ? CModelNames::ResolveModelID(std::get<std::string>(variantModelID)) : std::get<unsigned int>(variantModelID);
```
EngineResetModelLODDistance

```suggestion
    unsigned short modelID;
    if (std::holds_alternative<std::string>(variantModelID))
        modelID = CModelNames::ResolveModelID(std::get<std::string>(variantModelID));
    else
        modelID = std::get<unsigned int>(variantModelID);
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -772,31 +772,21 @@ int CLuaEngineDefs::EngineSetModelLODDistance(lua_State* luaVM)
-            return 1;
-        }
-    }
+    //Make sure we're dealing with a valid LOD distance, and not setting the same LOD distance
```
EngineResetModelLODDistance
```suggestion
    // Make sure we're dealing with a valid LOD distance, and not setting the same LOD distance
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -772,31 +772,21 @@ int CLuaEngineDefs::EngineSetModelLODDistance(lua_State* luaVM)
-    }
+    //Make sure we're dealing with a valid LOD distance, and not setting the same LOD distance
+    const float originalDistance = modelInfo->GetOriginalLODDistance();
+    if (originalDistance == 0.0f || originalDistance == modelInfo->GetLODDistance())
```
Does the new parser throw an error if a signed number is given to an unsigned type?

qaisjp commented on Server/mods/deathmatch/logic/luadefs/CLuaPlayerDefs.cpp:
```diff
@@ -946,47 +946,24 @@ int CLuaPlayerDefs::TakePlayerScreenShot(lua_State* luaVM)
-        if (bTemp)
-            argStream.SetCustomError("You can only pass false to hide the debug window or a level (0-3)");
+        if (std::get<bool>(variant))
+            throw std::invalid_argument("You can only pass false to turn the debugger off, or a level(0 - 3)");
```
SetPlayerScriptDebugLevel

```suggestion
            throw std::invalid_argument("You can only pass false to hide the debug window or a level (0-3)");
```

Tbh this false thing is weird - should get rid of the false thing in a separate PR.

qaisjp commented on Server/mods/deathmatch/logic/luadefs/CLuaPlayerDefs.h:
```diff
@@ -66,7 +66,8 @@ class CLuaPlayerDefs : public CLuaDefs
     LUA_DECLARE(DetonateSatchels);
     LUA_DECLARE(TakePlayerScreenShot);
-    LUA_DECLARE(SetPlayerScriptDebugLevel);
+
```
SetPlayerScriptDebugLevel
```suggestion
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp:
```diff
@@ -1762,19 +1762,13 @@ int CLuaPedDefs::SetPedDoingGangDriveby(lua_State* luaVM)
-        return luaL_error(luaVM, argStream.GetFullErrorMessage());
+    // Is valid style?
+    if (style < 4 || style > 16)
+        throw std::invalid_argument("Style can only be between 4 and 16");
```
Before: silent failure
After: error

Should this be a warning?

qaisjp commented on Client/mods/deathmatch/logic/lua/CLuaManager.cpp:
```diff
@@ -8,6 +8,8 @@
  *  Multi Theft Auto is available from http://www.multitheftauto.com/
  *
  *****************************************************************************/
+#include "../logic/lua/CLuaFunctionParser.h"
+#include "../logic/luadefs/CLuaDefs.h"
```
CLuaColshapeDefs

```suggestion
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -379,109 +354,109 @@ int CLuaColShapeDefs::CreateColTube(lua_State* luaVM)
 }
 
-int CLuaColShapeDefs::IsInsideColShape(lua_State* luaVM)
+int CLuaColShapeDefs::GetColShapeType(lua_State* luaVM)
```
Has this and some other functions been moved around or something? Please undo

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -558,144 +533,67 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
-        return luaL_error(luaVM, argStream.GetFullErrorMessage());
+    const auto colPolygon = static_cast<CClientColPolygon*>(colShape);
+    if (colPolygon->CountPoints() <= 3)
+        throw std::invalid_argument("Can't remove the last 3 points");
```
Should be warning, not error

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -558,144 +533,67 @@ int CLuaColShapeDefs::GetColPolygonPoints(lua_State* luaVM)
-        return 1;
-    }
+    if (!CStaticFunctionDefinitions::RemoveColPolygonPoint(colPolygon, pointIndex - 1))
+        throw std::invalid_argument("Invalid point index");
```
Should be warning, not error

qaisjp commented on Server/mods/deathmatch/logic/luadefs/CLuaPlayerDefs.cpp:
```diff
@@ -1493,33 +1493,6 @@ unsigned int CLuaPlayerDefs::GetPlayerScriptDebugLevel(CPlayer* const player)
-        lua_pushboolean(luaVM, false);
-        return 1;
-    }
-}
```
Needs fixup ?

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp:
```diff
@@ -1100,7 +1100,7 @@ bool CLuaEngineDefs::EngineSetModelVisibleTime(const std::variant<std::string, u
 
     if (hourOff > 24 || hourOff < 0)
-        throw std::invalid_argument("hour on must be between 0 and 24");
+        throw std::invalid_argument("hour off must be between 0 and 24");
```
This commit needs a fixup

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -541,7 +541,7 @@ const CVector2D& CLuaColShapeDefs::GetColPolygonPointPosition_OOP(CClientColShap
     if (colShape->GetShapeType() == COLSHAPE_POLYGON)
     {
-        pointIndex--; // Lua indexing starts at 1, we start at 0.
+        pointIndex--;            // Lua indexing starts at 1, we start at 0.
```
colshape
```suggestion
        // Lua indexing starts at 1, we start at 0
        pointIndex--;
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -555,8 +555,8 @@ bool CLuaColShapeDefs::SetColPolygonPointPosition(CClientColShape* const colShap
-    
-    pointIndex--; // lua starts at 1, we at 0
+
+    pointIndex--;            // lua starts at 1, we at 0
```
colshape
```suggestion
    // Lua indexing starts at 1, we start at 0.
    pointIndex--;
```

qaisjp commented on Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp:
```diff
@@ -586,7 +586,7 @@ bool CLuaColShapeDefs::RemoveColPolygonPoint(CClientColShape* const colShape, un
         throw std::invalid_argument("ColShape must be polygon");
 
-    pointIndex--; // lua starts at 1, we at 0
+    pointIndex--;            // lua starts at 1, we at 0
```
colshape
```suggestion
    // Lua indexing starts at 1, we start at 0.
    pointIndex--;
```

qaisjp reviewed (changes requested):
Annotated each suggested change with the name of the commit it should be melded into, so try and batch matching names if you accept the changes.

Pirulax committed:
Add ResolveModelID(const std::variant<string, ushort>&) to CStaticFDs. Resolves the given variant to a modelID

Pirulax commented:
https://github.com/multitheftauto/mtasa-blue/pull/1454#discussion_r429587512
@qaisjp 
I cant comment on the issue, but no, its I've deleted the old function, and added the new, one liner(CStaticFDs was always returning true, so there was no need for that check)

Pirulax commented:
Its still not compiling because we need to resolve the `dxDrawWiredSphere` argument `SColorARGB`. What should we do with that? Should I revert that commit? Or we implement SColorARGB?

@sbx320 I think I need your opinion here as well. I've asked about it before, but we didnt resolve it.

qaisjp commented:
Implement _SColor_ in a separate PR the same way the old parser works.

Pirulax commented:
Alright, alright, there we go.
I did it this time.

qaisjp commented:
Also, I didn't mention it earlier when I worked with you on the rebase since I didn't want to spend too much time with you watching rewords, but you need to reword every commit to be more clear.

"Refactor X" is not clear enough - someone must be able to read the commit message and know exactly what change was made.

It should be "Refactor X to use new Lua arg parser", and if there are other changes, include those in the commit descriptions.

Pirulax commented:
Okay, no problem.
Ill do it tomorrow.

Pirulax commented:
3261f1f, 6ed8668, 629e84e are messed up. I'm not quite sure how to fix it. I've googled it up, but the instructions on StackOverflow aren't totally clear to me, so if you could please explain it to me how could I change the contents of those commits, so they aren't totally messed up? 

What I have in mind is to just cherry pick those 3 commits into a separate branch, there you can then separately commit each change (there a feature in VSC where you can select the lines, and make a commit out of the selected lines), and then just merge that branch back into this one.

if that makes sense.

Pirulax commented:
Actually i could just go to another branch, squash merge the whole branch, and stage-commit changes line-by-line, so every commit would be perfect.

If theres a way to merge without having a 'merged branch..' commit automatically added?

Pirulax commented:
Im not quite sure whats up with the new parser?
Please take a look at the build logs

Pirulax commented:
All refactors have been moved to other PRs.
